main/src/main/scala/sbt/Defaults.scala

```diff
@@ -508,18 +509,18 @@ object Defaults extends BuildCommon {
       (cp, s) =>
         val ans: Seq[Analysis] = cp.flatMap(_.metadata get Keys.analysis) map { case a0: Analysis => a0 }
         val succeeded = TestStatus.read(succeededFile(s.cacheDirectory))
-        val stamps = collection.mutable.Map.empty[File, Long]
+        val stamps = collection.mutable.Map.empty[String, Long]
```

Comment:
Stamps were tracked per source file but now are tracked per class (that is named by a String). Tracking stamps by class name is more natural when everything else is tracked by class name (as it is in zinc).